### DISCUSSION TITLE
How to we get query two join tables in one query? Or how do we query a second join table with an array of ids?

### DISCUSSION THREAD

**Discussion by fergusmeiklejohn** [supabase/supabase#2468] [Questions] [answered]
I'm querying a join table `event_presentations` to return an array of objects. Each object has an id which I then want to use to query another join table `presentation_speakers` which would return an arrays of different lengths of objects. So the returned JSON would be like this:
```js
[
{
parent_id: 1234,
children: [{...}, {...}, {...}]
},
{
parent_id: 4567,
children: [{...}, {...}]
}
]
```
This is the supabase query to fetch the presentations:
```javascript
  const { data, error } = await supabase
    .from("event_presentations")
    .select(
      `presentations(
        presentation_id,
        title
      ),
      scheduled,
      timezone,
      session_name,
      keynote`
    )
    .eq("event_id", eventId);
  if (error) {
    console.log("error accessing database: ", error);
  }
  if (data) {
    return data
  }
```

This is the supabase query to fetch each speaker associated with a presentation:
```javascript
  const { data, error } = await supabase
    .from("presentation_speakers")
    .select(
      `presentation_id,
      speakers(
        speaker_id,
        full_name,
        title,
        avatar,
        organisation(name, logo),
      )
      `
    )
    .eq("presentation_id", presentationId);
```

QUESTION: how to get the presentations and speakers in one query? Can we use embedding? https://supabase.io/docs/reference/javascript/limit#with-embedded-resources Or if one query is impossible, how do we efficiently query a join table with an array of ids?

This is the schema:
![Screenshot 2021-07-20 at 17 57 35](https://user-images.githubusercontent.com/11864025/126408619-00cb0b74-3656-410d-95b3-b958f34a1a8b.png)

**Comment by steve-chavez** [comment, marked as answer]
> QUESTION: how to get the presentations and speakers in one query? Can we use embedding?

Yes, according to your schema, presentations and speakers form a many-to-many relationship through the `presentations_speakers` junction. So you should be able to do this:

```js
  const { data, error } = await supabase
    .from("presentations")
    .select(`
      id,
      speakers(
        speaker_id,
        full_name,
        title,
        avatar,
        organisation(name, logo),
      )`
    )
    .eq("id", presentationId);
```

**Reply by fergusmeiklejohn** [reply]
Thanks Steve. For complex queries found it easier to work in Datagrip to get the SQL right, then create a View which I can query easily from the client
